ui/plugins/eu.esdihumboldt.hale.ui.functions.numeric/src/eu/esdihumboldt/hale/ui/functions/numeric/SequentialIDParameterPage.java

```diff
@@ -235,10 +239,43 @@ public void modifyText(ModifyEvent e) {
 			infoDeco.setShowOnlyOnFocus(true);
 		}
 
+		// specify startValue for the sequence generation
+		if (getParametersToHandle().containsKey(START_VALUE)) {
+			label = new Label(page, SWT.NONE);
+			label.setText("StartValue");
+			labelLayout.applyTo(label);
+
+			startValue = new Text(page, SWT.SINGLE | SWT.BORDER);
+			controlLayout.applyTo(startValue);
+
+			startValue.setText(
+					getOptionalInitialValue(START_VALUE, new ParameterValue("1")).as(String.class));
+
+			startValue.addModifyListener(new ModifyListener() {
+
+				@Override
+				public void modifyText(ModifyEvent e) {
+					updateStatus();
```

Comment:
Would you expect any validation of what was entered?

Reply:
I honestly copied it from the above paragraph ("specify suffix") with a basic understanding of what is happening here. 

I would not expect a validation on the integer value because the int can be anything. But I would expect it to be an integer and nothing else.

What should I do?

Reply:
Right now the user should enter an integer or use a project variable.
But they can enter any string - so it would be quite easy to enter something that is not valid and leads to an error.

Doing some kind of validation (e.g. further below where the example is generated) could help - but I think it is also fine to not do it - then the user just needs to deal with the potential transformation error.

Another way to support the user would be to make sure that if they use a wrong value (which you can't for sure check in the function editor always, as they could use project variables), the error message from the transformation is in such a way that it is easily clear to users what the problem is and how they can solve it.

Reply:
I've tried out my solution for the validation of `startValue`, please check it in my latest commit. My logic:

- in `SequentialIDParameterPage.java`, `startValue` is of type string; I don't want to modify its type and I don't see an easy way to validate the content here, for a string that - for its intrinsic nature  - can be anything;
- I rather think that the validation can happen where the string is parsed into an integer, that is in `SequentialID.java`;
- I use the logic `try{} catch() {}` to modify the error message;
- I believe that the error can only be generated by the user inputting a string that does not represent an integer (any sequence of chars or a float given as a string)
- I verified that - if in the app one gives a sequence of chars or a float number - then one gets an `Instance validation warning` with the wished message.